node/sc/bridge_manager.go

```diff
@@ -203,41 +211,49 @@ func (bi *BridgeInfo) GetCounterPartToken(token common.Address) common.Address {
 	return cpToken
 }
 
-func (bi *BridgeInfo) GetPendingRequestEvents() []*RequestValueTransferEvent {
+//func (bi *BridgeInfo) GetPendingRequestEvents() []*RequestValueTransferEvent {
+func (bi *BridgeInfo) GetPendingRequestEvents() []interface{} {
 	ready := bi.pendingRequestEvent.Pop(maxPendingNonceDiff / 2)
-	var readyEvent []*RequestValueTransferEvent
-	for _, item := range ready {
-		readyEvent = append(readyEvent, item.(*RequestValueTransferEvent))
+	readyEvent := make([]interface{}, len(ready))
+	for i, item := range ready {
+		readyEvent[i] = item
 	}
-
 	vtPendingRequestEventCounter.Dec((int64)(len(ready)))
-
 	return readyEvent
 }
 
 // processingPendingRequestEvents handles pending request value transfer events of the bridge.
-func (bi *BridgeInfo) processingPendingRequestEvents() error {
+func (bi *BridgeInfo) processingPendingRequestEvents() {
 	ReadyEvent := bi.GetReadyRequestValueTransferEvents()
 	if ReadyEvent == nil {
-		return nil
+		return
 	}
 
 	logger.Trace("Get ready request value transfer event", "len(readyEvent)", len(ReadyEvent), "len(pendingEvent)", bi.pendingRequestEvent.Len())
 
-	for idx, ev := range ReadyEvent {
-		if ev.RequestNonce < bi.lowerHandleNonce || bi.handledEvent.Exist(ev.RequestNonce) {
-			logger.Trace("handled requests can be ignored", "RequestNonce", ev.RequestNonce, "lowerHandleNonce", bi.lowerHandleNonce)
-			continue
+	for idx, _ev := range ReadyEvent {
+		switch ev := _ev.(type) {
+		case *RequestValueTransferEvent:
```

Comment:
It seems that we can make an interface to make this simple...

Reply:
What interface do you mean? The event struct is already interfaced.

Reply:
Interface of RequestValueTransferEvent and RequestValueTransferEncodedEvent.
Then, I think we don't need this switch statement.

Reply:
The concrete type conversion (type assertion) must proceed before do member access.

Reply:
Yeah you can do that using the interface.

Reply:
1. The event is interfaced.
2. The concrete type conversion is with switch stmt.

If my comments didn't answer your question, please describe what you are seeing in detail.